tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/DebugWriter.java

```diff
@@ -0,0 +1,126 @@
+ * logging behaviour to help avoid the need for ad-hoc logging via {@code System.out}.
+ *
+ * <p>In most cases a mapping class can just have a {@code main} method like:
+ * <pre>{@code
+ *   // Arguments: <output-dir> [<log-level>]
+ *   public static void main(String[] args) throws IOException {
+ *       DebugWriter.writeForDebugging(args, MapperClass::process);
+ *   }
+ * }</pre>
+ *
+ * <p>Note however that running this still requires that the {@code CLDR_DIR} system
+ * property is set.
+ */
+public final class DebugWriter {
+    private static final String PACKAGE_ROOT = "org.unicode.icu.tool.cldrtoicu";
+    private static final String PACKAGE_PREFIX = PACKAGE_ROOT + ".";
+
+    /**
+     * Writes the IcuData generated by the given function using the default {@code CLDR_DIR}
+     * system property.
+     *
+     * <p>This is a helper method to make it easy for each mapper to have its own main
+     * method for debugging, and it should not be used directly by {@code LdmlConverter}.
+     */
+    public static void writeForDebugging(String[] args, Function<CldrDataSupplier, IcuData> fn)
+            throws IOException {
+        writeMultipleForDebugging(args, src -> ImmutableList.of(fn.apply(src)));
+    }
+
+    /**
+     * Writes the IcuData generated by the given function using the default {@code CLDR_DIR}
+     * system property.
+     *
+     * <p>This is a helper method to make it easy for each mapper to have its own main
+     * method for debugging, and it should not be used directly by {@code LdmlConverter}.
+     */
+    public static void writeMultipleForDebugging(
+            String[] args, Function<CldrDataSupplier, Collection<IcuData>> fn)
+            throws IOException {
+        String cldrPath = System.getProperty("CLDR_DIR", System.getenv("CLDR_DIR"));
+        checkState(cldrPath != null,
+                "required 'CLDR_DIR' system property or environment variable not set");
+        checkArgument(args.length >= 1, "expected output directory");
+        Path outDir = Paths.get(args[0]);
+        String logLevel = (args.length == 2) ? args[1] : "OFF";
+
+        String loggerConfig = Joiner.on("\n").join(
+                "handlers = java.util.logging.ConsoleHandler",
+                "java.util.logging.ConsoleHandler.level     = ALL",
+                "java.util.logging.ConsoleHandler.encoding  = UTF-8",
+                "java.util.logging.ConsoleHandler.formatter = " + LogFormatter.class.getName(),
+                "",
+                PACKAGE_ROOT + ".level  = " + logLevel);
+        LogManager.getLogManager()
+                .readConfiguration(new ByteArrayInputStream(loggerConfig.getBytes(UTF_8)));
+
+        Files.createDirectories(outDir);
+        CldrDataSupplier src = CldrDataSupplier.forCldrFilesIn(Paths.get(cldrPath));
+        for (IcuData icuData : fn.apply(src)) {
+            IcuTextWriter.writeToFile(icuData, outDir, ImmutableList.of("Debug output"), true);
```

Comment:
pull the ImmutableList.of() out of the loop?

Reply:
True.